tests/unittests/test_datasource/test_azure_helper.py

```diff
@@ -778,6 +1145,67 @@ def test_instantiates_shim_with_kwargs(self, shim):
             shim.call_args)
 
 
+class TestGetMetadataGoalStateXMLAndReportFailureToFabric(CiTestCase):
```

Comment:
The UTs in the class **above** (`TestGetMetadataGoalStateXMLAndReportReadyToFabric`) test report **ready** functionality.
The UTs in **this** class (`TestGetMetadataGoalStateXMLAndReportFailureToFabric`) test report **failure** functionality.
The UTs in this class are the report failure UT equivalents of the class above.